core/indexer/processTransactions.go

```diff
@@ -136,6 +141,16 @@ func (tdp *txDatabaseProcessor) prepareTransactionsForDatabase(
 	return append(convertMapTxsToSlice(transactions), rewardsTxs...), alteredAddresses
 }
 
+func (tdp *txDatabaseProcessor) addScrsReceiverToAlteredAccounts(alteredAddress map[string]struct{}, scrs map[string]*smartContractResult.SmartContractResult) {
```

Comment:
no need to inject shardCoordinator, could have used the current shard ID and a range on the provided body. Miniblocks already contain destination/sender shard ID

Reply:
It does not work as the intrashard miniblock holding smart contract results is not added in the body.